docs/source/guides/inference.mdx

```diff
@@ -1,58 +1,314 @@
+example, we generated an image from a text prompt. The returned value is a `PIL.Image` object that can be saved to a
+file.
+
+### Using a specific model
+
+What if you want to use a specific model? You can specify it either as a parameter or directly at an instance level:
+
+```python
+>>> from huggingface_hub import InferenceClient
+# Initialize client for a specific model
+>>> client = InferenceClient(model="prompthero/openjourney-v4")
+>>> client.text_to_image(...)
+# Or use a generic client but pass your model as an argument 
+>>> client = InferenceClient()
+>>> client.text_to_image(..., model="prompthero/openjourney-v4")
+```
+
+<Tip>
+
+There are more than 200k models on the Hugging Face Hub! Each task in the [`InferenceClient`] comes with a recommended
+model. However in most cases you'll be interested in finding a model specific to _your_ needs. Visit the
+https://huggingface.co/models page to explore your possibilities. 
+
+</Tip>
+
+### Using a specific URL
+
+The examples we saw above are using the free-hosted Inference API. This proves to be very useful to quickly prototype
+and test thins. Once you're ready to deploy your model to production, you'll need to use a dedicated infrastructure.
+That's were [Inference Endpoints](https://huggingface.co/inference-endpoints) comes into play. It allows you to deploy
+any model and expose it as a private API. Once deployed, you'll get a URL that you can connect to using exactly the same
+code as before, changing only the `model` parameter:
+
+```python
+>>> from huggingface_hub import InferenceClient
+>>> client = InferenceClient(model="https://uu149rez6gw9ehej.eu-west-1.aws.endpoints.huggingface.cloud/deepfloyd-if")
+# or 
+>>> client = InferenceClient()
+>>> client.text_to_image(..., model="https://uu149rez6gw9ehej.eu-west-1.aws.endpoints.huggingface.cloud/deepfloyd-if")
+```
+
+### Authentication
+
+Calls made with the [`InferenceClient`] can be authenticated using a [User Access Token](https://huggingface.co/docs/hub/security-tokens).
+By default, it will use the token saved on your machine if you are logged in (check out
+[how to login](https://huggingface.co/docs/huggingface_hub/quick-start#login)). If you are not logged in, you can pass
+your token as an instance parameter:
+
+```python
+>>> from huggingface_hub import InferenceClient
+>>> client = InferenceClient(token="hf_***")
+```
+
+<Tip>
+
+Authentication is NOT mandatory when using the Inference API. However, authenticated users gets a higher free-tier to
+play with the service. Token is also mandatory if you want to run inference on your private models or on private
+endpoints.
+
+</Tip>
+
+## Supported tasks
+
+[`InferenceClient`]'s goal is to provide the most easy-to-use interface to run inference on Hugging Face models. It
+has a simple API that supports the most common tasks. Here is a list of the currently supported tasks:
+
+- audio
```

Comment:
Good idea. I updated it in https://github.com/huggingface/huggingface_hub/pull/1474/commits/978fbf29ea7107772d9484f0d47fa4572aab8fad. I removed the "models" columns with number of models since it would need to be manually updated and I'm pretty sure it will not :smile: Having a link to the hf.co/tasks page is really nice though.